website/src/components/Toc.vue

```diff
@@ -26,11 +26,45 @@ export default {
     filteredHeadings() {
       return this.headings.filter(heading => heading.level < 4)
     }
+  },
+
+  watch: {
+    $route() {
+      this.isRoute = true
+      this.currentHash = this.$route.hash
+    }
+  },
+
+  data() {
+    return {
+      currentHash: null,
+      observer: null,
+      isRoute: false,
+      justMounted: true
+    }
+  },
+  
+  mounted() {
+    this.observer = new IntersectionObserver(([firstEntry]) => {
+      if (this.isRoute || this.justMounted) {
+      	this.isRoute = false
+      	this.justMounted = false
+      } else if (firstEntry.boundingClientRect.bottom <= firstEntry.intersectionRect.bottom) {
+        const hash = `#${firstEntry.target.id}`
+        history.pushState(null, null, hash)
```

Comment:
Let's remove this, it will make your browser history a mess by just scrolling in the page.

Reply:
I will change it to replace

Reply:
I think that's unnecessary either.

Reply:
so there is no need to change the hash? Ok

Reply:
We do need to change the hash, I guess you're right, `.replace` should work.

Reply:
Oops, you were too fast 😄

Reply:
No problem